fix(apiservernames): use apiservernnames field

FalcoSuessgott opened the pull request:
Faced a  bug where I wasn't able to set `apiservernnames`, because the wrong attribute was read during create.

I noticed you use a lot of `TypeSet` for Lists of String. I would rather use `TypeList` for these Attributes or Limit `TypeSet` with `MaxItem: 1`.  Otherwise there is a chance that the Items in the Set have duplicate elements.

scott-the-programmer commented on minikube/resource_cluster.go:
```diff
@@ -271,13 +267,18 @@ func initialiseMinikubeClient(d *schema.ResourceData, m interface{}) (service.Cl
 		return nil, err
 	}
 
+	apiserverNames := []string{}
+	if d.Get("apiserver_names").(*schema.Set).Len() > 0 {
+		apiserverNames = append(apiserverNames, d.Get("apiserver_names").(*schema.Set).List()[0].(string))
+	}
+
```
It appears that apiserver_names can be multiple values i.e.

```console
minikube start --apiserver-names "foo.minikube.com" --apiserver-names "bar.minikube.com"
```

So I think we should keep the property as it is schema_cluster.go and instead 

```suggestion
	apiserverNames := []string{}
	if d.Get("apiserver_names").(*schema.Set).Len() > 0 {
		apiserverNames = d.Get("apiserver_names")
	}
```

and then cast it using the `ReadSliceState` function `APIServerNames:  state_utils.ReadSliceState(apiserverNames)` 

In terms of preferring a  `Set` over a `List`. This is mostly so we can avoid causing an accidental `diff` where the minikube cli would otherwise no-op. I think it's largely OK to allow duplicates to be configured, as we ultimately rely on what the underlying minikube lib allows (which does in fact to allow duplicate apiserver names apparently 😕)

scott-the-programmer reviewed (commented):
Thanks for spotting this and submitting a PR! I've been swamped lately, so any help is appreciated ❤️ 

Your changes LGTM, 1 minor comment around allowing multiple values to be configured. Can we also scope this PR to just the resource__cluster.go changes? 

We can probably follow up with a wider project lint 👍🏼

FalcoSuessgott commented:
Hi @scott-the-programmer,

no problem happy to help. I dropped the gofumpt commit and removed `MaxItem` from the schema.

I made some testing with the resulting provider and it seems to work:) :

```hcl
terraform {
  required_version = ">= 1.5.0"

  required_providers {
    minikube = {
      source  = "scott-the-programmer/minikube"
      version = "0.3.4"
    }
  }
}

resource "minikube_cluster" "docker" {
  driver       = "docker"
  cluster_name = "vault-playground"
  cni          = "bridge"

  listen_address = "0.0.0.0"
  
  apiserver_names = [
    "host.docker.internal",
  ]

  addons = [
    # "dashboard",
    "default-storageclass",
    "storage-provisioner"
  ]
}
```

Cert Details:
![image](https://github.com/scott-the-programmer/terraform-provider-minikube/assets/18051472/13f7ee43-620f-4bad-9fda-bcd485488cd4)
